packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1323,7 +1343,13 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
      * then retries them in a case of an error, etc.
      */
     protected async _runTaskFunction() {
-        const source = this.requestQueue || this.requestList || (await this.getRequestQueue());
+        // Ensure we have a request provider
+        if (!this.requestProvider) {
+            this.requestQueue = await this.getRequestQueue();
+            this.requestProvider = this.requestQueue;
+        }
```

Comment:
Why not use the `initializeRequestProvider` function? By calling it, we ensure that `this.requestProvider` is defined, right?